### PR TITLE
New to old sorting

### PR DESCRIPTION
**Pull request by khl02007**
Final addition (hopefully) to `old_api_utils`: added a class to make an old sorting extractor from a new sorting. Need this because old sorting doesn't have segments and some method names are different (e.g. old `get_units_spike_train` vs. new `get_all_spike_trains`). Mostly copied from `spikeextractors` repo.

### CODE REVIEW

File: spikeinterface/core/old_api_utils.py [hunk 1]
```diff
@@ -45,16 +45,171 @@ def get_channel_property(self, channel_id, property):
...
+
+    def get_unit_ids(self):
+        """This function returns a list of ids (ints) for each unit in the sorsted result.
+
+        Returns
+        -------
+        unit_ids: array_like
+            A list of the unit ids in the sorted result (ints).
+        """
+        return self._sorting.get_unit_ids()
```
Comment (alejoe91) [inline spikeinterface/core/old_api_utils.py:74]: In the new API, unit ids can also be str (or any object), while in the old API they're expected to be int. So here we have to check: if they are int, propagate them, if they are not int then they should be reset to int

**Review by alejoe91 — CHANGES_REQUESTED** [review]

File: spikeinterface/core/old_api_utils.py (hunk 1 above)
Comment (alejoe91) [reply spikeinterface/core/old_api_utils.py]: @khl02007 I tried to fix the int issue. Let me know what you think!

**Comment by khl02007** [comment]
Thanks for that. My goal was to allow other functions to treat this extractor like an old api extractor. In that case they would expect the unit ids of this extractor to be int not str. So if the unit id of the original (new api) sorting is not int or cannot be converted to int, then I think trying to instantiate this class should just raise an error and say that the new sorting cannot be converted to an old extractor without first redefining the unit ids. What do you think?

**Comment by khl02007** [comment]
Actually, thinking through this again I think it may be useful to just convert the unit ids to int automatically as you have done, but maybe print a warning so that the user knows what happened. Other than that, feel free to merge.

File: spikeinterface/core/old_api_utils.py [hunk 2]
```diff
@@ -51,6 +51,176 @@ def create_extractor_from_new_recording(new_recording):
...
+    def set_sampling_frequency(self, sampling_frequency):
+        """
+        It sets the sorting extractor sampling frequency.
+
+        Parameters
+        ----------
+        sampling_frequency: float
+            The sampling frequency
+        """
+        self._sampling_frequency = sampling_frequency
```
Comment (samuelgarcia) [inline spikeinterface/core/old_api_utils.py:165]: I would raise an error here.
We can't change the sampling rate on the fly in the new API.

File: spikeinterface/core/old_api_utils.py [hunk 3]
```diff
@@ -51,6 +51,176 @@ def create_extractor_from_new_recording(new_recording):
...
+
+        Parameters
+        ----------
+        times: array-like
+            The times in seconds for each frame
+        """
+        max_frames = np.array([np.max(self.get_unit_spike_train(u)) for u in self.get_unit_ids()])
+        assert np.all(max_frames < len(times)), "The length of 'times' should be greater than the maximum " \
+                                                "spike frame index"
+        self._times = times.astype('float64')
```
Comment (samuelgarcia) [inline spikeinterface/core/old_api_utils.py:178]: Same here. times is not handle in the new API. I would raise an error.

Comment (samuelgarcia) [reply spikeinterface/core/old_api_utils.py:178]: Unless we are OK that this times do not propagate to the new API, it is only something bound to this object.

**Comment by samuelgarcia** [comment]
Hi Kyu.
Thanks for this. I made some comments.

File: spikeinterface/core/old_api_utils.py (hunk 2 above)
Comment (alejoe91) [reply spikeinterface/core/old_api_utils.py:165]: or remove completely

File: spikeinterface/core/old_api_utils.py (hunk 3 above)
Comment (khl02007) [reply spikeinterface/core/old_api_utils.py]: Done

File: spikeinterface/core/old_api_utils.py (hunk 2 above)
Comment (khl02007) [reply spikeinterface/core/old_api_utils.py]: Removed

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**